### PR TITLE
feat: Hotfit for Nitro loading on CPU with hyper-threading support

### PR DESCRIPTION
**Pull request by vuonghoainam**
Problem statement:
- Nitro needs to know exactly the number of physical core and can combine it to run with `numactl` on linux

Solution:
- Add `systeminformation` to get cpu.physicalCores as INT an pass to `nitro`

- [x] Mac M
- [x] Windows
- [x] Linux 

References:
- Based on https://github.com/sebhildebrandt/systeminformation?tab=readme-ov-file#3-cpu
CleanShot 2023-12-11 at 22 02 29@2x

### CODE REVIEW

**Comment by linhtran174** [comment]
Model cannot be loaded on my Windows PC, how do you check the logging when spawning Nitro process?

File: extensions/inference-nitro-extension/src/module.ts [hunk 1]
```diff
@@ -191,9 +198,13 @@ async function spawnNitroProcess(): Promise<void> {
     const binaryPath = path.join(binaryFolder, binaryName);
 
     // Execute the binary
-    subprocess = spawn(binaryPath, [1, "127.0.0.1", PORT], {
-      cwd: binaryFolder,
-    });
+    subprocess = spawn(
+      binaryPath,
+      [nitroResourceProbe.numCpuPhysicalCore, "127.0.0.1", PORT],
```
Comment (tikikun) [inline extensions/inference-nitro-extension/src/module.ts]: The number here always 1

**Review by tikikun — APPROVED** [review]
lgtn

**Comment by linhtran174** [comment]
lgtm

**Pull request merged**